src/linter/index.tsx

```diff
@@ -0,0 +1,78 @@
+type Message = {
+  column: number | null;
+  description: string | string[];
+  file: string;
+  line: number | null;
+  message: string;
+  type: 'notice' | 'error' | 'warning';
+  uid: string;
+};
+
+export type ExternalMessage = Message & {
+  // These are some extra properties that we don't need to work with.
+  context: string[];
+  for_appversions: object;
+  id: string[];
+  tier: number;
+};
+
+type ExternalLinterResult = {
+  error: null | string;
+  full_report_url: string;
+  upload: string;
+  url: string;
+  validation: {
+    detected_type: string;
+    ending_tier: number;
+    errors: number;
+    message_tree: object;
+    messages: ExternalMessage[];
+    metadata: object;
+    notices: number;
+    success: boolean;
+    warnings: number;
+  };
+};
+
+const createInternalMessage = (message: ExternalMessage): Message => {
+  return {
+    column: message.column,
+    description: message.description,
+    file: message.file,
+    line: message.line,
+    message: message.message,
+    type: message.type,
+    uid: message.uid,
+  };
+};
+
+type MessageMap = {
+  [relativeFileName: string]: {
+    global: Message[];
+    byLine: { [line: number]: Message[] };
```

Comment:
I wonder how we'll consume this map in the UI 🤔  Either we'll test `byLine` for each line or do an `Object.keys()` before the loop over all the lines of the file. I am just thinking out loud.

Reply:
I have a feeling we will adjust the map shape once we start using it but that's OK. It's harder to guess but I didn't want to wait on the API.